File: gitlab-nic/code-samples-main/API/Elixir/info-authcode.exs
# Requests an authorization code. 

url = "https://api-sandbox.nic.fr/v1/registrar/authorization-code-requests"

if length(System.argv()) < 1 do
  raise "Boom Usage: program request-ID"
end

request = String.trim(List.first(System.argv()))
result = HTTPoison.get!(url <> "/" <> URI.encode(request), Afnic.get_headers(), [])

case result do
  %HTTPoison.Response{status_code: 200, body: body} ->
    response = Jason.decode!(body)
    IO.puts("State of #{request}: #{response["status"]} (last update #{response["updateDate"]})")

    if response["status"] == "ACCEPTED" do
      IO.puts("The authorization code for \"#{response["domainName"]}\" is #{response["code"]}")
    end

  %HTTPoison.Response{status_code: 400, body: body} ->
    response = Jason.decode!(body)
    IO.puts("No info for #{request} because #{List.first(response["errors"])["message"]}")

  other ->
    raise "Boom (#{inspect(other)})"
end

File: gitlab-nic/code-samples-main/API/Python/info-authcode.py
#!/usr/bin/env python3

# Gets information about an authorization code. 

import requests
import json
import sys

import afnic

if len(sys.argv) != 2:
    raise Exception("Usage: %s request-ID" % sys.argv[0])
requestId = sys.argv[1]
headers = afnic.get_headers()
response = requests.get("https://api-sandbox.nic.fr/v1/registrar/authorization-code-requests/%s" % requestId,
                         headers=headers)
body = json.loads(response.text)
if response.status_code != 200:
    raise Exception("HTTP status code for this demand is %s (reason \"%s\")" % (response.status_code, body))
print("Request %s for domain %s is in state %s" % (body["repositoryObjectId"], body["domainName"], body["status"]))
if "code" in body:
    print("Authorization code is %s (valid until %s)" % (body["code"], body["codeExpirationDate"]))
    
      


    
